core/src/main/java/com/dtolabs/rundeck/core/encrypter/PasswordUtilityEncrypterPlugin.java

```diff
@@ -0,0 +1,8 @@
+package com.dtolabs.rundeck.core.encrypter;
+
+import java.util.Map;
+
+public interface PasswordUtilityEncrypterPlugin {
+
+    Map encrypt(Map config);
```

Comment:
this interface is a bit crude for a plugin type, I would suggest something more like this:

```suggestion
    EncryptorResponse encrypt(Map<String,String> config);
```

where the `EncryptorResponse` is an interface like:

~~~
interface EncryptorResponse{
    boolean isValid()
    String getError()
    Map<String,String> getOutputs()
}
~~~

Reply:
added @gschueler